src/main/java/org/sagebionetworks/web/client/widget/header/Header.java

```diff
@@ -50,32 +57,45 @@ public Widget asWidget() {
 		view.setPresenter(this);
 		return view.asWidget();
 	}
-	
+
 	public void setSearchVisible(boolean searchVisible) {
 		view.setSearchVisible(searchVisible);
 	}
-	
+
 	public void refresh() {
+		UserSessionData userSessionData = authenticationController.getCurrentUserSessionData();
+		if (cachedUserSessionData == null || cachedUserSessionData.equals(userSessionData)) {
+			cachedUserSessionData = userSessionData;
+			view.setUser(userSessionData);
+		}
+		if (userSessionData != null) {
+			initUserFavorites(new Callback(){
+				@Override
+				public void invoke() {
+					// only get called on failure
+					view.clearFavorite();
+				}
+			});
+		}
 		view.refresh();
 		view.setSearchVisible(true);
 	}
```

Comment:
Please review refresh()
case 1, both cachedUserSessionData & userSessionData are null => view.setUser()
case 2, cachedUserSessionData is null and userSessionData is not => view.setUser() then initUserFavorites()
case 3, cachedUserSessionData is not null and userSessionData is null => view.setUser() then do not initUserFavorites()
case 4, both are not null & different => view.setUser() then initUserFavorites()
case 5, both are not null & same => initUserFavorites()